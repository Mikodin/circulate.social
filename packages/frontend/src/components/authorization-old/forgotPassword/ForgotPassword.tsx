import { useState, Fragment } from 'react';
import { useRouter } from 'next/router';
import { Form, Input, Button, Alert } from 'antd';
import { MailOutlined, LockOutlined } from '@ant-design/icons';

import { AUTH_FORMS } from '../AuthContainer';
import css from './ForgotPassword.module.scss';

export interface Props {
  fetchInitForgotPassword: (username: string) => Promise<boolean>;
  fetchFinalizeForgotPassword: (
    username: string,
    newPassword: string,
    code: string
  ) => Promise<boolean>;
  // eslint-disable-next-line
  onSuccess?: (result?: any) => void;
  showForm?: (form: AUTH_FORMS) => void;
  updateSeedValues?: (userValues: { email?: string }) => void;
  seedEmail?: string;
  redirectTo?: string;
}

const ForgotPassword = (props: Props): JSX.Element => {
  const router = useRouter();
  const [form] = Form.useForm();

  const {
    fetchInitForgotPassword,
    fetchFinalizeForgotPassword,
    redirectTo,
    onSuccess,
    seedEmail,
  } = props;
  const [isInvalidCredentials, setIsInvalidCredentials] = useState(false);
  const [showConfirmationCode, setShowConfirmationCode] = useState(false);
  const [showLimitError, setShowLimitError] = useState(false);
  const [isLoginInFlight, setIsLoginInFlight] = useState(false);

  interface FormValues {
    email: string;
    newPassword: string;
    confirmationCode: string;
  }
  const onFinish = async (values: FormValues): Promise<void> => {
    const { email, newPassword, confirmationCode } = values;
    try {
      let result;
      setIsLoginInFlight(true);
      if (showConfirmationCode) {
        result = await fetchFinalizeForgotPassword(
          email,
          confirmationCode,
          newPassword
        );

        if (redirectTo) {
          router.push(redirectTo);
        }

        if (onSuccess) {
          onSuccess(result);
        }
      } else {
        result = await fetchInitForgotPassword(email);
        setShowConfirmationCode(true);
      }

      setIsLoginInFlight(false);
    } catch (error) {
      if (error && error.code === 'LimitExceededException') {
        setShowLimitError(true);
      }
      if (error && error.code === 'ExpiredCodeException') {
        setIsInvalidCredentials(true);
      }
      if (error && error.code === 'CodeMismatchException') {
        setIsInvalidCredentials(true);
      }

      throw error;
    }
  };

  return (
    <Fragment>
      <Form
        className={css.form}
        form={form}
        name="horizontal_login"
        layout="vertical"
        onFinish={onFinish}
        initialValues={{
          email: seedEmail || undefined,
          password: undefined,
          newPassword: undefined,
          confirmationCode: undefined,
        }}
        onValuesChange={({ email }): void => props.updateSeedValues({ email })}
      >
        <Form.Item
          name="email"
          rules={[{ required: true, message: 'Please input your email' }]}
          label="Email Address"
        >
          <Input
            prefix={<MailOutlined className="site-form-item-icon" />}
            autoComplete="email"
            placeholder="joedoe@gmail.com"
            type="email"
          />
        </Form.Item>
        {showConfirmationCode && (
          <Fragment>
            <Form.Item
              name="newPassword"
              rules={[
                { required: true, message: 'Please input a new password' },
              ]}
              label="New password"
            >
              <Input
                prefix={<LockOutlined className="site-form-item-icon" />}
                autoComplete="new-password"
                type="password"
                placeholder="Password"
              />
            </Form.Item>
            <Form.Item
              name="confirmationCode"
              rules={[
                {
                  required: true,
                  message: 'Please input the code emailed to you',
                },
              ]}
              label="Confirmation Code"
            >
              <Input
                prefix={<LockOutlined className="site-form-item-icon" />}
                autoComplete="one-time-code"
                placeholder="Password"
              />
            </Form.Item>
            <a
              onClick={(): void => {
                fetchInitForgotPassword(form.getFieldValue('email'));
              }}
            >
              {"Didn't receive the code? Resend it"}
            </a>
          </Fragment>
        )}
        {isInvalidCredentials && (
          <Alert
            message="Invalid verification code provided, please try again."
            type="error"
            showIcon
          />
        )}
        {showLimitError && (
          <Alert
            message="Attempt limit exceeded, please try after some time."
            type="error"
            showIcon
          />
        )}

        <Form.Item shouldUpdate={true}>
          {(): JSX.Element => {
            const isEmailTouched =
              form.isFieldTouched('email') ||
              Boolean(form.getFieldValue('email'));
            // const isNewPasswordTouched = form.isFieldTouched('newPassword');
            const isFormTouched = isEmailTouched;
            return (
              <Button
                data-testid="submitButton"
                loading={isLoginInFlight}
                type="primary"
                htmlType="submit"
                size="large"
                block
                disabled={Boolean(
                  !isFormTouched ||
                    form.getFieldsError().filter(({ errors }) => errors.length)
                      .length
                )}
              >
                {isLoginInFlight ? 'Setting New Password' : 'Set New Password'}
              </Button>
            );
          }}
        </Form.Item>
      </Form>
    </Fragment>
  );
};

export default ForgotPassword;
