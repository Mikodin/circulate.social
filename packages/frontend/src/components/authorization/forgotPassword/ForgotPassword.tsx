import { useState, Fragment } from 'react';
import { Form, Input, Button, Alert } from 'antd';
import { MailOutlined, LockOutlined } from '@ant-design/icons';

import { AUTH_FORMS } from '../AuthContainer';
import css from './ForgotPassword.module.scss';

export interface FormValues {
  email: string;
  password: string;
  confirmationCode: string;
}
export interface Props {
  fetchInitForgotPassword: (username: string) => Promise<boolean>;
  fetchFinalizeForgotPassword: (
    username: string,
    code: string,
    password: string
  ) => Promise<boolean>;
  // eslint-disable-next-line
  showForm?: (form: AUTH_FORMS) => void;
  updateSeedValues?: (userValues: {
    email?: string;
    password?: string;
  }) => void;
  seedEmail?: string;
  onFormCompletionCallback: (formValues: Partial<FormValues>) => Promise<void>;
}

const ForgotPassword = (props: Props): JSX.Element => {
  const [form] = Form.useForm();

  const {
    fetchInitForgotPassword,
    fetchFinalizeForgotPassword,
    seedEmail,
    onFormCompletionCallback,
  } = props;
  const [isCodeMismatchError, setIsCodeMismatchError] = useState(false);
  const [isExpiredCodeError, setIsExpiredCodeError] = useState(false);
  const [showLimitError, setShowLimitError] = useState(false);
  const [isInvalidParameterError, setIsInvalidParameterError] = useState(false);

  const [showConfirmationCode, setShowConfirmationCode] = useState(false);
  const [isForgotPasswordInFlight, setIsForgotPasswordInFlight] = useState(
    false
  );

  const onFinish = async (values: FormValues): Promise<void> => {
    const { email, password, confirmationCode } = values;
    try {
      setIsForgotPasswordInFlight(true);

      if (showConfirmationCode) {
        await fetchFinalizeForgotPassword(email, confirmationCode, password);
        onFormCompletionCallback({ email, password });
      } else {
        await fetchInitForgotPassword(email);
        setShowConfirmationCode(true);
      }

      setShowLimitError(false);
      setIsInvalidParameterError(false);
      setIsCodeMismatchError(false);
      setIsForgotPasswordInFlight(false);
    } catch (error) {
      setIsForgotPasswordInFlight(false);
      console.error(error);
      if (error && error.code === 'LimitExceededException') {
        setShowLimitError(true);
      }
      if (error && error.code === 'ExpiredCodeException') {
        setIsExpiredCodeError(true);
      }
      if (error && error.code === 'CodeMismatchException') {
        setIsCodeMismatchError(true);
      }
      if (error && error.code === 'InvalidParameterException') {
        setIsInvalidParameterError(true);
      }
    }
  };

  return (
    <Fragment>
      <Form
        className={css.form}
        form={form}
        name="horizontal_login"
        size="large"
        layout="vertical"
        onFinish={onFinish}
        initialValues={{
          email: seedEmail || undefined,
          password: undefined,
          confirmationCode: undefined,
        }}
        onValuesChange={({ email, password, confirmationCode }): void => {
          if (password) {
            setIsInvalidParameterError(false);
          }
          if (confirmationCode) {
            setIsCodeMismatchError(false);
          }
          props.updateSeedValues({ email, password });
        }}
      >
        <Form.Item
          name="email"
          rules={[{ required: true, message: 'Please input your email' }]}
          label="Email Address"
        >
          <Input
            prefix={<MailOutlined className="site-form-item-icon" />}
            autoComplete="email"
            placeholder="joedoe@gmail.com"
            type="email"
          />
        </Form.Item>
        {showConfirmationCode && (
          <Fragment>
            <Form.Item
              name="password"
              rules={[
                { required: true, message: 'Please input a new password' },
              ]}
              label="New password"
            >
              <Input.Password
                visibilityToggle
                prefix={<LockOutlined className="site-form-item-icon" />}
                autoComplete="new-password"
                type="password"
                placeholder="Password"
              />
            </Form.Item>
            <Form.Item
              name="confirmationCode"
              rules={[
                {
                  required: true,
                  message: 'Please input the code emailed to you',
                },
              ]}
              label="Confirmation Code"
            >
              <Input
                prefix={<LockOutlined className="site-form-item-icon" />}
                autoComplete="one-time-code"
                placeholder="123456"
              />
            </Form.Item>
            <a
              onClick={(): void => {
                fetchInitForgotPassword(form.getFieldValue('email'));
              }}
            >
              {isForgotPasswordInFlight
                ? 'Sending...'
                : "Didn't receive the code? Resend it"}
            </a>
          </Fragment>
        )}
        {isCodeMismatchError && (
          <Alert
            message="Invalid verification code provided, please try again."
            type="error"
            showIcon
          />
        )}
        {showLimitError && (
          <Alert
            message="Attempt limit exceeded, please try after some time."
            type="error"
            showIcon
          />
        )}
        {isExpiredCodeError && (
          <Alert
            message="The code provided has expired.  Please try again."
            type="error"
            showIcon
          />
        )}
        {isInvalidParameterError && (
          <Alert
            message="Your password is too weak. It must have atleast 8 characters, a capital letter, a number, and a symbol."
            type="error"
            showIcon
          />
        )}
        <Form.Item shouldUpdate={true}>
          {(): JSX.Element => {
            const isEmailTouched =
              form.isFieldTouched('email') ||
              Boolean(form.getFieldValue('email'));
            // const isNewPasswordTouched = form.isFieldTouched('newPassword');
            const isFormTouched = isEmailTouched;
            return (
              <Button
                data-testid="submitButton"
                loading={isForgotPasswordInFlight}
                type="primary"
                htmlType="submit"
                size="large"
                block
                disabled={Boolean(
                  !isFormTouched ||
                    form.getFieldsError().filter(({ errors }) => errors.length)
                      .length
                )}
              >
                {isForgotPasswordInFlight ? 'Submitting...' : 'Submit'}
              </Button>
            );
          }}
        </Form.Item>
      </Form>
    </Fragment>
  );
};

export default ForgotPassword;
