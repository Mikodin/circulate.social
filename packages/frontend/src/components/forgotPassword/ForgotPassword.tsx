import { useState, Fragment, useContext, useEffect } from 'react';
import { useRouter } from 'next/router';
import { Form, Input, Button, Alert } from 'antd';
import { MailOutlined, LockOutlined } from '@ant-design/icons';

import { AUTH_FORMS } from '@components/authContainer/AuthContainer';
import UserContext from '../../state-management/UserContext';
import css from './forgotPassword.module.scss';

interface Props {
  seedEmailAddress?: string;
  redirectTo?: string;
  onSuccess?: (result?: any) => any;
  showForm?: (form: AUTH_FORMS) => any;
}
const ForgotPassword = (props: Props) => {
  const router = useRouter();
  const { forgotPasswordInit, forgotPasswordSubmit } = useContext(UserContext);
  const [form] = Form.useForm();

  const { redirectTo, onSuccess, seedEmailAddress, showForm } = props;
  const [isInvalidCredentials, setIsInvalidCredentials] = useState(false);
  const [showConfirmationCode, setShowConfirmationCode] = useState(false);
  const [showLimitError, setShowLimitError] = useState(false);
  const [isLoginInFlight, setIsLoginInFlight] = useState(false);

  const onFinish = async (values) => {
    const { email, newPassword, confirmationCode } = values;
    try {
      let result;
      if (showConfirmationCode) {
        result = await forgotPasswordSubmit(
          email,
          confirmationCode,
          newPassword
        );

        if (redirectTo) {
          router.push(redirectTo);
        }

        if (onSuccess) {
          onSuccess(result);
        }
      } else {
        result = await forgotPasswordInit(email);
        setShowConfirmationCode(true);
      }
    } catch (error) {
      if (error && error.code === 'LimitExceededException') {
        setShowLimitError(true);
      }
      if (error && error.code === 'ExpiredCodeException') {
        setIsInvalidCredentials(true);
      }
      if (error && error.code === 'CodeMismatchException') {
        setIsInvalidCredentials(true);
      }

      throw error;
    }
  };

  return (
    <Fragment>
      <Form
        className={css.form}
        form={form}
        name="horizontal_login"
        layout="vertical"
        onFinish={onFinish}
        initialValues={{
          email: seedEmailAddress || undefined,
          newPassword: undefined,
          confirmationCode: undefined,
          password: undefined,
        }}
      >
        <Form.Item
          name="email"
          rules={[{ required: true, message: 'Please input your email' }]}
          label="Email Address"
        >
          <Input
            prefix={<MailOutlined className="site-form-item-icon" />}
            placeholder="johndoe@gmail.com"
            type="email"
          />
        </Form.Item>
        {showConfirmationCode && (
          <Fragment>
            <Form.Item
              name="newPassword"
              rules={[
                { required: true, message: 'Please input a new password' },
              ]}
              label="New password"
            >
              <Input
                prefix={<LockOutlined className="site-form-item-icon" />}
                type="password"
                placeholder="Password"
              />
            </Form.Item>
            <Form.Item
              name="confirmationCode"
              rules={[
                {
                  required: true,
                  message: 'Please input the code emailed to you',
                },
              ]}
              label="Confirmation Code"
            >
              <Input
                prefix={<LockOutlined className="site-form-item-icon" />}
                type="password"
                placeholder="Password"
              />
            </Form.Item>
            <a
              onClick={() => {
                forgotPasswordInit(form.getFieldValue('email'));
              }}
            >
              Didn't receive the code? Resend it
            </a>
          </Fragment>
        )}
        {isInvalidCredentials && (
          <Alert
            message="Invalid verification code provided, please try again."
            type="error"
            showIcon
          />
        )}
        {showLimitError && (
          <Alert
            message="Attempt limit exceeded, please try after some time."
            type="error"
            showIcon
          />
        )}

        <Form.Item shouldUpdate={true}>
          {() => {
            const isEmailTouched =
              form.isFieldTouched('email') ||
              Boolean(form.getFieldValue('email'));
            const isNewPasswordTouched = form.isFieldTouched('newPassword');
            const isFormTouched = isEmailTouched;
            return (
              <Button
                loading={isLoginInFlight}
                type="primary"
                htmlType="submit"
                size="large"
                block
                disabled={Boolean(
                  !isFormTouched ||
                    form.getFieldsError().filter(({ errors }) => errors.length)
                      .length
                )}
              >
                {isLoginInFlight ? 'Setting New Password' : 'Set New Password'}
              </Button>
            );
          }}
        </Form.Item>
      </Form>
    </Fragment>
  );
};

export default ForgotPassword;
