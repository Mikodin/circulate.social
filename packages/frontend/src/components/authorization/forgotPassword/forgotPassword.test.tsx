import React from 'react';
import {
  render,
  RenderResult,
  fireEvent,
  waitFor,
  act,
} from '@testing-library/react';

import ForgotPassword, { Props } from './ForgotPassword';

const defaultProps = {
  seedEmail: '',
  redirectTo: '',
  onSuccess: jest.fn(),
  showForm: jest.fn(),
  updateSeedValues: jest.fn(),
  fetchInitForgotPassword: jest.fn(),
  fetchFinalizeForgotPassword: jest.fn(),
  onFormCompletionCallback: jest.fn(),
};

function renderForgotPassword(props?: Partial<Props>): RenderResult {
  return render(<ForgotPassword {...defaultProps} {...props} />);
}

describe('ForgotPassword', () => {
  it('should render', () => {
    const { container } = renderForgotPassword();
    expect(container).toMatchSnapshot();
  });

  describe('When given seedEmail props ', () => {
    describe('When props.seedEmail is populated', () => {
      it('Should be prefilled with the props.seedEmail', () => {
        const seedEmail = 'mike@circulate.social';
        const { queryByDisplayValue } = renderForgotPassword({
          seedEmail,
        });
        const input = queryByDisplayValue(seedEmail);
        expect(input).toBeTruthy();
      });
    });
  });

  describe('Events on input', () => {
    describe('When email is inputted', () => {
      it('should call updateSeedValues with the inputted email', () => {
        const { queryByPlaceholderText } = renderForgotPassword();
        const input = queryByPlaceholderText('joedoe@gmail.com');

        const inputtedEmail = 'mike@circulate.social';
        fireEvent.change(input, { target: { value: inputtedEmail } });

        expect(defaultProps.updateSeedValues).toHaveBeenCalledWith({
          email: inputtedEmail,
          password: undefined,
        });
      });
    });
  });

  describe('On submit of a complete form', () => {
    const fetchInitForgotPasswordSpy = jest.fn(() => Promise.resolve(true));
    const fetchFinalizeForgotPasswordSpy = jest.fn(() => Promise.resolve(true));
    const onFormCompletionCallbackSpy = jest.fn();
    beforeEach(() => {
      fetchInitForgotPasswordSpy.mockClear();
      fetchFinalizeForgotPasswordSpy.mockClear();
    });

    const setupCompleteForm = async (
      waitForFetchInitForgotPasswordToFinish: boolean
    ): Promise<RenderResult> => {
      const container = renderForgotPassword({
        seedEmail: 'aasdf',
        fetchInitForgotPassword: fetchInitForgotPasswordSpy,
        fetchFinalizeForgotPassword: fetchFinalizeForgotPasswordSpy,
        onFormCompletionCallback: onFormCompletionCallbackSpy,
      });
      const { queryByTestId, queryByPlaceholderText } = container;
      const emailInput = queryByPlaceholderText('joedoe@gmail.com');

      const submitButton = queryByTestId('submitButton');

      if (waitForFetchInitForgotPasswordToFinish) {
        await act(async () => {
          fireEvent.change(emailInput, {
            target: { value: 'mike@circulate.social' },
          });

          fireEvent.submit(submitButton);
        });
      } else {
        act(() => {
          fireEvent.change(emailInput, {
            target: { value: 'mike@circulate.social' },
          });

          fireEvent.submit(submitButton);
        });
      }

      return container;
    };

    describe('On submit of first step just email form', () => {
      it('should call props.fetchInitForgotPasswordSpy with the email', async () => {
        await setupCompleteForm(false);
        await waitFor(() =>
          expect(fetchInitForgotPasswordSpy).toHaveBeenCalledWith(
            'mike@circulate.social'
          )
        );
      });

      it('Should display "Submitting..." when the request is in flight', async () => {
        const { queryByText } = await setupCompleteForm(false);
        await waitFor(() => expect(queryByText(/Submitting.../i)).toBeTruthy());
      });
    });

    describe('On step 2 confirming code and setting new password', () => {
      const emailInputValue = 'mike@circulate.social';
      const newPasswordValue = 'newPass1!';
      const confirmationCodeValue = '09876';

      const setupCompleteStep2Form = async (): Promise<RenderResult> => {
        const step2FormContainer = await setupCompleteForm(true);
        const { queryByTestId, queryByPlaceholderText } = step2FormContainer;
        const newPasswordInput = queryByPlaceholderText('Password');
        const confirmationCodeInput = queryByPlaceholderText('123456');
        const submitButton = queryByTestId('submitButton');

        await act(async () => {
          fireEvent.change(newPasswordInput, {
            target: { value: newPasswordValue },
          });
          fireEvent.change(confirmationCodeInput, {
            target: { value: confirmationCodeValue },
          });

          fireEvent.submit(submitButton);
        });
        return step2FormContainer;
      };

      it('Should display the new password and confirmation code form with email already filled', async () => {
        const {
          queryByPlaceholderText,
          queryByDisplayValue,
        } = await setupCompleteForm(true);

        expect(queryByDisplayValue(emailInputValue)).toBeTruthy();
        expect(queryByPlaceholderText('Password')).toBeTruthy();
        expect(queryByPlaceholderText('123456')).toBeTruthy();
      });

      describe('On submit of a filled new password and confirmation form', () => {
        it('Should fire fetchFinalizeForgotPassword with the email, new password and confirmation code', async () => {
          await setupCompleteStep2Form();

          await waitFor(() =>
            expect(fetchFinalizeForgotPasswordSpy).toHaveBeenCalledWith(
              emailInputValue,
              confirmationCodeValue,
              newPasswordValue
            )
          );
        });
      });

      it('Should fire props.onFormCompletionCallback with the email, and new password', async () => {
        await setupCompleteStep2Form();
        await waitFor(() =>
          expect(onFormCompletionCallbackSpy).toHaveBeenCalledWith({
            email: emailInputValue,
            password: newPasswordValue,
          })
        );
      });
    });
  });

  describe('On an incomplete form', () => {
    const fetchInitForgotPasswordPromise = Promise.resolve(true);
    const fetchInitForgotPasswordSpy = jest.fn(
      () => fetchInitForgotPasswordPromise
    );

    beforeEach(() => {
      fetchInitForgotPasswordSpy.mockClear();
    });

    const setupIncompleteForm = (fieldsToUpdate: 'email'[]): RenderResult => {
      const container = renderForgotPassword({
        seedEmail: '',
        fetchInitForgotPassword: fetchInitForgotPasswordSpy,
      });

      const { queryByPlaceholderText } = container;

      act(() => {
        fieldsToUpdate.forEach((fieldName) => {
          let textToQueryFor;

          switch (fieldName) {
            case 'email':
              textToQueryFor = 'joedoe@gmail.com';
              break;
            default:
              break;
          }
          const input = queryByPlaceholderText(textToQueryFor);

          fireEvent.change(input, {
            target: { value: 'SomeValueThatDoesntMatter' },
          });
        });
      });

      return container;
    };

    describe('When email field is empty', () => {
      it('should not call props.fetchRegister', async () => {
        const { queryByTestId } = setupIncompleteForm([]);

        const submitButton = queryByTestId('submitButton');
        fireEvent.submit(submitButton);
        await waitFor(() =>
          expect(fetchInitForgotPasswordSpy).not.toHaveBeenCalled()
        );
      });

      it('should display "Please input your email"', async () => {
        const { queryByText, queryByTestId } = setupIncompleteForm([]);
        const submitButton = queryByTestId('submitButton');
        fireEvent.submit(submitButton);

        await waitFor(() =>
          expect(queryByText(/Please input your email/i)).toBeTruthy()
        );
      });
    });
  });

  // TODO Return to this
  describe.skip('When API requests go wrong', () => {
    const setupFormAfterFirstSubmit = async (
      fetchInitForgotPasswordMockOverride?: jest.Mock<Promise<boolean>, []>
    ): Promise<RenderResult> => {
      const fetchInitForgotPasswordMock = jest.fn(() => Promise.resolve(true));
      const container = renderForgotPassword({
        seedEmail: 'aasdf',
        fetchInitForgotPassword:
          fetchInitForgotPasswordMock || fetchInitForgotPasswordMockOverride,
      });
      const { queryByTestId, queryByPlaceholderText } = container;
      const emailInput = queryByPlaceholderText('joedoe@gmail.com');

      const submitButton = queryByTestId('submitButton');

      act(() => {
        fireEvent.change(emailInput, {
          target: { value: 'mike@circulate.social' },
        });

        fireEvent.submit(submitButton);
      });

      return container;
    };

    describe('When the user tries to forgotPassword too many times', () => {
      it('Should display a limit error to the use', async () => {
        const limitExceededExceptionMock = jest.fn(() =>
          Promise.reject({ code: 'LimitExceededException' })
        );
        const { queryByText } = await setupFormAfterFirstSubmit(
          limitExceededExceptionMock
        );

        await waitFor(() => {
          expect(limitExceededExceptionMock).toBeCalledWith('as');
          expect(
            queryByText(/Attempt limit exceeded, please try after some time./)
          ).toBeTruthy();
        });
      });
    });
    describe('First form error states', () => {
      it('Should display a limit error to the use', async () => {
        const limitExceededExceptionMock = jest.fn(() =>
          Promise.reject({ code: 'CodeMismatchException' })
        );
        const { queryByText } = await setupFormAfterFirstSubmit(
          limitExceededExceptionMock
        );

        await waitFor(() => {
          expect(limitExceededExceptionMock).toBeCalledWith('as');
          expect(
            queryByText(/Attempt limit exceeded, please try after some time./)
          ).toBeTruthy();
        });
      });
    });
  });
});
