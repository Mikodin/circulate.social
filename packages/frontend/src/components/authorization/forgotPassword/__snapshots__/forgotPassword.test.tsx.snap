// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword should render 1`] = `
<div>
  <form
    class="ant-form ant-form-vertical ant-form-hide-required-mark ant-form-large form"
    id="horizontal_login"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="horizontal_login_email"
          title="Email Address"
        >
          Email Address
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <span
              class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
            >
              <span
                class="ant-input-prefix"
              >
                <span
                  aria-label="mail"
                  class="anticon anticon-mail site-form-item-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="mail"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                    />
                  </svg>
                </span>
              </span>
              <input
                autocomplete="email"
                class="ant-input ant-input-lg"
                id="horizontal_login_email"
                placeholder="joedoe@gmail.com"
                type="email"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
              data-testid="submitButton"
              disabled=""
              type="submit"
            >
              <span>
                Submit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
